### PR TITLE
Lua display and code reduction potpourri 

### LINKED ISSUE #840: Lua V2 display problem on small displays
## Current Behavior

The elrsv2.lua from the current master branch has some display problems on the black and white 128x64 screens:
IMG_20210826_110309035
 

## Steps to Reproduce

1. Use radio with black and white 128x64 screen
2. Flash current master TX module firmware
3. Open elrsV2.lua script
4. See the black bars in the above photo

## Your Environment

- TX hardware: DIY 2400 internal T-Lite module
- RX hardware: -
- Handset model: Jumper T-Lite
- OpenTX version (including nightly number): 2.3.12N503
- ExpressLRS version (TX & RX MUST MATCH): ver.unknown:1dad58
- Packet Rate: -
- Telemetry Ratio: -
- user_defines:

```
-DMY_BINDING_PHRASE="notgonnatellyou"
-DRegulatory_Domain_ISM_2400
-DENABLE_TELEMETRY
-DHYBRID_SWITCHES_8
-DFEATURE_OPENTX_SYNC
-DUART_INVERTED
-DAUTO_WIFI_ON_INTERVAL=60
-DHOME_WIFI_SSID="notgonnatellyou"
-DHOME_WIFI_PASSWORD="lol"
-DUSE_DYNAMIC_POWER
```

### LINKED ISSUE #852: Incorrect use of +/- buttons on TX in LUA script.
## Current Behavior

You have to press '-' to *increase* packet rate and power.

## Steps to Reproduce

Press '+' to increase rate or power, note that it does nothing.

## Possible Solution (Not obligatory)

Swap the button handling code or something, so that '+' increases things and '-' decreases things.

## Details

N/A

## Your Environment

EdgeTX 2.4.0, ELRS 1.0.1.

### PR DESCRIPTION
**Pull request by CapnBry**
Compared to my usual concise PR statement, this PR is more of a potpourri of things somewhat related to Lua. 436 bytes of flash saved, 20 bytes of RAM, this is a mix of prettification and code prettification changes.

* Remove static lua device structure that was just taking up space. Send it with the new generic `packetQueueExtended()` CRSF packet generator. We also now send "ELRS" as our serial number in the device structure to help identify ourselves. There's space (4 bytes) for version number there too, we should take advantage. Saves 20 RAM, 72 flash
* Convert `sendELRSstatus()` to generic. Saves 280 flash
* Convert `sendSyncPacketToTX()` to generic. Saves 116 flash
* Fix HappyModel ES24TX displaying as DIY
* Fix CAPS in TX module display names
* Fix "no double confirmation" for non-ELRS lua users (most times)
* Adds HooJ folder item for TBS Agent-Lite compatibility
* Lua display changes (in TX firmware)
  * Air rate options now ordered slowest->fastest. Fixes #852
  * Ratio / replaced with :
  * P.Rate becomes Packet Rate
  * T.Rate becomes Telem Ratio
  * VTX ADMINISTRATOR -> VTX Administrator
* Lua display changes (in Lua)
  * Script name displayed in OpenTX now "ExpressLRS"
  * Commands get [brackets] and are no longer inverted on B&W
  * Folders get > (arrow?) and are bold B&W instead of inverted. Also are drawn with their own handler
  * Right justify the good/bad in the title bar and use the rest of the space for the device name
  * Shift COL2 right on 128x64 by a few pixels
  * Fix Int type field displays being too far right on 128x64
  * Fixes #840
  BW128x64 lua display

### CODE REVIEW

**Comment by CapnBry** [comment]
Github, you are smoking crack.

File: src/lua/color_display_480x272/TOOLS/elrsV2.lua [hunk 1]
```diff
@@ -459,6 +463,14 @@ local function refreshNext()
   end
 end
 
+local function lcd_title()
+  lcd.clear()
+  lcd.drawFilledRectangle(0, 0, LCD_W, 30, TITLE_BGCOLOR)
+  local title = (allParamsLoaded == 1 or elrsFlags > 0) and deviceName or "Loading..."
+  lcd.drawText(1, 5, title, MENU_TITLE_COLOR)
```
Comment (pkendall64) [inline src/lua/color_display_480x272/TOOLS/elrsV2.lua:470]: Should just be `lcd.drawText(1, 5, title)` as you are writing in the title line and the colors are automatically used

File: src/lua/color_display_480x272/TOOLS/elrsV2.lua [hunk 2]
```diff
@@ -459,6 +463,14 @@ local function refreshNext()
...
+  lcd.drawText(LCD_W, 5, tostring(badPkt) .. "/" .. tostring(goodPkt), MENU_TITLE_COLOR + RIGHT)
```
Comment (pkendall64) [inline src/lua/color_display_480x272/TOOLS/elrsV2.lua:471]: Likewise, `lcd.drawText(LCD_W, 5, tostring(badPkt) .. "/" .. tostring(goodPkt), RIGHT)`

File: src/lua/color_display_480x272/TOOLS/elrsV2.lua (hunk 1 above)
Comment (CapnBry) [reply src/lua/color_display_480x272/TOOLS/elrsV2.lua]: But... why... why does that work?! Updated with your suggestion and it works as expected. Thanks so much!

**Review by maybenikhil — APPROVED** [review]

**Comment by CapnBry** [comment]
@StonedDawg I like the consistency in the lua script for the folder display, perhaps we can be a little more descriptive in the cancel messafges?
* "Reboot to stop wifi"
* "Reboot to stop BT"

Both of these fit on the 128x64 using this capitalization and I think provide more insight than the generic "cancel" message.

**Comment by StonedDawg** [comment]
@CapnBry yeah of course, as long as it fits to the smallest screen, its better to be more descriptive, and we should not worry about how this messages appear in non ELRS lua script since the 2nd popup's message doesn't show up anyway.

the double OK sometimes failed on me when we press OK/Enter too fast (2nd confirmation message is not loaded yet, therefore the module is still in its previous state), is this also what happened to you?

**Comment by CapnBry** [comment]
> the double OK sometimes failed on me when we press OK/Enter too fast (2nd confirmation message is not loaded yet, therefore the module is still in its previous state), is this also what happened to you?

I'm really not sure what causes it. I tried it a dozen times and couldn't find a pattern. When I got the double OK it seemed to be more with if I waited for the first message to go away automatically, but not every time. If I pressed OK when it was still on the first message, I think it would usually skip the confirmation (since I already pressed OK) and go right to the REBOOT message. 

With the standard CRSF Configuration, it just says "Proceed WiFi Update" and then the whole configuration engine is locked up. Pressing OK does nothing, pressing BACK does nothing. Holding BACK exits the script and then the configuration doesn't work at all any more "Waiting for Crossfire" and that's it.

**Comment by StonedDawg** [comment]
uhm yes, with the standard CRSF configurator (or deviationTX), we don't do double confirmation because : 

1. the popup's message doesn't always show up anyway.
2. once the wifi starts, the popup will be locked up and will not show the next message, it might confuse user if we ask for a confirmation but after user gives input, the popup doesn't change.
3.  our LUA script is actually showing up the "current" messages, from 1 packet sent previously. this way we can display "reboot to cancel" while actually our module is not responding to any crsf UART anymore.

what you are having with standard CRSF configuration is probably wifi mode has started.

**Comment by CapnBry** [comment]
Yup, you're 100% right. It was in wifi update mode so that's why it wouldn't do anything after.

Yeah so it all looks good to me if you just want to change those two reboot messages to just give a little extra information for polish.

**Comment by CapnBry** [comment]
I've tested pkendall64's rebase and it all looks good to me. I stamp this as ready to merge.

**Review by StonedDawg — APPROVED** [review]

**Review by pkendall64 — APPROVED** [review]

**Review by maybenikhil — APPROVED** [review]

**Comment by hydra** [comment]
Does the updated script work with v1.x TX's as well as later code or ?

**Comment by pkendall64** [comment]
> Does the updated script work with v1.x TX's as well as later code or ?

No, this v2 script is only for the current master branch as the LUA code on the TX module was completely reworked.

**Pull request merged**